addons/web/static/src/search/search_model.js

```diff
@@ -939,10 +940,97 @@ export class SearchModel extends EventBus {
         this._notify();
     }
 
+    /**
+     * Because it require a RPC call to get the properties search items,
+     * it's done lazily, only when we need them.
+     *
+     * @param {Object} searchItem
+     * @param {string} definitionRecordModel
+     * @param {string} definitionRecordField
+     */
+    async getSearchItemsProperty(searchItem, definitionRecord, definitionRecordModel, definitionRecordField) {
+        const result = await this._fetchPropertiesDefinition(
+            definitionRecordModel, definitionRecordField);
+
+        const searchItems = [];
+
+        for (const [definitionRecordId, definitionRecordName, definitions] of result) {
+            for (const definition of definitions) {
+                const item = Object.values(this.searchItems).find(item =>
+                    item.fieldName === searchItem.fieldName
+                    && item.propertyName === definition.name);
+
+                if (item) {
+                    // already in the list, can happen if we unfold the properties field
+                    // open a form view, edit the property and then go back to the search view
+                    // the label of the property might have been changed
+                    item["description"] = `${definition.string} (${definitionRecordName})`;
+                    searchItems.push(item);
+                    continue;
+                }
+
+                const operator = ["many2many", "tags"].includes(definition.type) ? "in" : undefined;
+
+                this.searchItems[this.nextId] = {
+                    "isActive": false,
+                    "type": "field_property",
+                    "fieldName": searchItem.fieldName,
+                    "propertyName": definition.name,
+                    "propertyDomain": [definitionRecord, "=", definitionRecordId],
+                    "fieldType": "properties",
+                    "description": `${definition.string} (${definitionRecordName})`,
+                    "groupId": this.nextGroupId,
+                    "id": this.nextId,
+                    "autocompleteValues": [],
+                    "definition": definition,
+                    "propertyItemId": searchItem.id,
+                    "relation": definition.comodel,
+                    "selection": definition.selection,
+                    "tags": definition.tags,
+                    "operator": operator,
```

Comment:
How about we put everything related to properties into a sub-structure to avoid polluting the main searchItem "specification" with a lot of keys?

So we would have:

```
{
    // all standard keys
    "fieldName": ...,
    "description": ...,
    "isPropertyField": true,
    "propertyFieldDefinition": {
        type: definition.type,
        relation: definition.comodel,
        ...
    }
}
```

That way :socks: are a bit better sorted?